lib/src/runner/browser/dartium.dart

```diff
@@ -150,8 +150,12 @@ class Dartium extends Browser {
     /// check whether it's actually connected to an isolate, indicating that
     /// it's the observatory for the main page. Once we find the one that is, we
     /// cancel the other requests and return it.
-    return (await inCompletionOrder(operations)
-        .firstWhere((url) => url != null, defaultValue: () => null)) as Uri;
+    try {
+      return (await inCompletionOrder(operations)
+          .firstWhere((url) => url != null)) as Uri;
+    } on StateError catch (_) {
+      return null;
```

Comment:
Add a comment about why you're doing `try`/`catch` rather than `orElse`.

Reply:
Done! Thanks

Reply:
wouldn't `var result = ...where((url) => url != null)); return result.isNotEmpty ? result.first : null;` be better than `try`/`catch`? I'd expect `isNotEmpty` only to look for the first result like `firstWhere` instead of materializing the whole result (but I haven't checked myself yet).

Reply:
You can't call multiple read methods on the same stream. `result.isNotEmpty` will create a subscription, which will make `result.first` fail.

Reply:
I see. Didn't realize it's a stream. Sorry for the noise.